libsolidity/ast/Types.cpp

```diff
@@ -669,7 +669,10 @@ tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal
 			// parse the exponent
 			bigint exp = bigint(string(expPoint + 1, _literal.value().end()));
 
-			if (exp > numeric_limits<int32_t>::max() || exp < numeric_limits<int32_t>::min())
+			// SOL-008: Huge values for exponent are causing a crash. Limiting maximum to rational(1) << 4096
```

Comment:
We could shorten this to something like `disallow gigantic exponents to limit memory consumption'.